prov/efa/src/rxr/rxr_rma.c

```diff
@@ -388,7 +388,8 @@ ssize_t rxr_rma_post_rtw(struct rxr_ep *ep, struct rxr_tx_entry *tx_entry)
 		return rxr_pkt_post_ctrl_or_queue(ep, RXR_TX_ENTRY, tx_entry, RXR_EAGER_RTW_PKT, 0);
 
 	if (tx_entry->total_len >= rxr_env.efa_max_long_write_size &&
-	    efa_ep_support_rdma_read(ep->rdm_ep) && rxr_peer_support_rdma_read(peer)) {
+	    efa_ep_support_rdma_read(ep->rdm_ep) &&
+	    (peer->is_self || rxr_peer_support_rdma_read(peer))) {
```

Comment:
The `peer->is_self` could go into `rxr_peer_support_rdma_read`.

Reply:
Sure. That would mean rxr_peer_support_rdma_read() need to have ```rxr_ep``` as an argument and will call ```efa_ep_support_rdma_read(rxr_ep)``` if ```peer->is_self```. Is that reasonable to you?

Reply:
On a second thought, it might be better to introduce ```rxr_both_support_rdma_read``` and fold the logic inside it.

Reply:
Thanks, this is cleaner.